ui.apps/src/main/content/jcr_root/apps/authoring-toolkit/depends-on/js/actions/dependsOnUpdateOptionsAction.js

```diff
@@ -0,0 +1,137 @@
+            });
+        });
+    }
+
+    /**
+     * Registers {@code update-datasource} action in DependsOn registry.
+     * This action performs updating of the option set of the current Select widget
+
+     * @param path      Path value as set in the foreign widget (a path picker) this widget depends on
+     * @param options   Options to be passed with an async HTTP request to the custom datasource (same as described
+     *                  in the {@code CustomDataSourceServlet} javadoc
+     */
+    DependsOn.ActionRegistry.register('update-options', function(path, options) {
+
+        // Initialize and check whether critical requisites are accessible; early return if not
+        const select = this.$el;
+
+        let resourceEndpoint = select.closest('form').attr('action') || '';
+        let valueMember = (select.attr('name') || '').replace(/^[./]+|\/+$/g, '');
+        if (valueMember.indexOf('/') > -1 && resourceEndpoint) {
+            resourceEndpoint += '/' + valueMember.substring(0, valueMember.lastIndexOf('/'));
+            valueMember = valueMember.substring(valueMember.lastIndexOf('/') + 1);
+        }
+        resourceEndpoint += '.json';
+
+        if (select[0].tagName !== 'CORAL-SELECT' || !path || resourceEndpoint === '.json' || !valueMember) {
+            return;
+        }
+
+        // Collect options to be passed to the custom datasource endpoint (path to the JCR node to get items from, etc.)
+        // and compose a HTTP query string
+        options.path = path;
+        options.output = 'json';
+
+        let datasourceEndpoint = Granite.HTTP.externalize('/apps/authoring-toolkit/datasources/option-provider');
+        $.each(options, function (key, value) {
+            datasourceEndpoint += (datasourceEndpoint.indexOf('?') > -1 ? '&' : '?') + key + '=' + value;
+        });
+
+        // Receive new options from the datasource; check if they are the same as options already present,
+        // and early return in such a case
+        $.get(datasourceEndpoint)
+            .then(function(newOptions) {
+                const existingOptions = select
+                    .find('coral-select-item')
+                    .map(function(index, item) {
+                        return {value: $(item).val()};
+                    })
+                    .toArray();
+                if (optionsAreSame(existingOptions, newOptions)) {
+                    return;
+                }
+
+                // If option set needs to be changed, cache the currently selected value for this widget from the underlying
+                // resource (so that a new option may be be assigned the "selected" state if its value matches),
+                // and then replace the option set in DOM
+                if (select.attr('data-stored-value')) {
+                    replaceOptions(select, newOptions, select.attr('data-stored-value'));
+                } else if (!select.attr('data-noCache')) {
```

Comment:
Are we sure that we can get camel-cased name part for attribute here ?

Reply:
Revised and removed this usage in general